src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Tables/ReferenceSearchParam.sql

```diff
@@ -6,6 +6,9 @@
     BaseUri                             varchar(128)            COLLATE Latin1_General_100_CS_AS NULL,
     ReferenceResourceTypeId             smallint                NULL,
     ReferenceResourceId                 varchar(64)             COLLATE Latin1_General_100_CS_AS NOT NULL,
+    CONSTRAINT PK_ReferenceSearchParam PRIMARY KEY NONCLUSTERED (ResourceTypeId, ResourceSurrogateId, SearchParamId, ReferenceResourceId)
```

Comment:
Is this guaranteed to be unique?

Reply:
Thanks Joyce to confirm. I found some gap in my analysis for this table and few other SearchParam tables as back in the time, I relied on the data in 1TB database and couple of resources on fhir spec. But I have now re-analyzed these tables and found some more work is required to be done on those since column that  need to be part of pk are NULLable and will be addressed in the next PR.

I have cleaned up this PR and it adds PK for 13 FHIR tables.